src/app/service/ndkprovider.service.ts

```diff
@@ -395,6 +391,38 @@ export class NdkproviderService {
     this.followedTopicsEmitter.emit(followedTopicsToPublish);
   }
 
+  async followUnfollowContact(hexPubKeyToFollow:string, follow:boolean){
+    var contactListFromCache = (await this.dbService.peopleIFollow.toArray()).map(item=>item.hexPubKey);
+
+    // if it is a follow event, attach the new contact to the list or else remove from the list if found and publish
+    if(follow){
+      contactListFromCache.push(hexPubKeyToFollow);
+    }else{
+      const index = contactListFromCache.indexOf(hexPubKeyToFollow);
+
+      if (index > -1) {
+        contactListFromCache.splice(index, 1);
+      }
+    }
+
+    var tags:NDKTag[] = contactListFromCache.map((item, index)=>{
+      return ["p", item.toString()]
+    });
+
+    const ndkEvent = new NDKEvent(this.ndk);
+    if(this.currentUser){
+      ndkEvent.pubkey = this.currentUser?.hexpubkey()
+    }
+
+    ndkEvent.tags = tags;
+    ndkEvent.kind = 3;
+    ndkEvent.content = "{\"wss://nos.lol\":{\"read\":true,\"write\":true}, \"wss://relay.nostr.band\":{\"read\":true,\"write\":true},\"wss://relay.f7z.io\":{\"read\":true,\"write\":true},\"wss://relay.damus.io\":{\"read\":true,\"write\":true},\"wss://nostr.mom\":{\"read\":true,\"write\":true}, \"wss://no.str.cr\":{\"read\":true,\"write\":true}}";
+    //TODO:fix this, generate it from relays array
+    ndkEvent.sign();
```

Comment:
Looks like the multiple auth requests are due to having sign and publish both. 

When we do sign and then publish, 2 auth requests come only on invocation of publish()

When i commented the sign(), it gave me only one auth request. Snort showed that person as unfollowed.  However zapddit still showed him as followed.

Why do we need sign here?

Reply:
apparently we need to await for the sign to complete, when we publish it after the signing completion it works without issues.

The reason we sign is we need the created_at to be populated